checker/src/org/checkerframework/checker/index/samelen/SameLenAnnotatedTypeFactory.java

```diff
@@ -214,4 +214,8 @@ public AnnotationMirror createSameLen(String... val) {
         builder.setValue("value", val);
         return builder.build();
     }
+
+    public AnnotationMirror createSameLenUnknown() {
```

Comment:
The `createSameLen(String...)` method creates a new object so this name makes sense.  This method does not create anything.   

> From the view of the user, they're receiving a fresh @SameLenUnknown annotation (but there's only one, so there's no need to create another, which I view as an implementation detail).

The method name leads the user to believe a fresh object is created each time, but it does not create a fresh object.  If you had named the methods getX, then it might be an implementation detail whether or not a new object is created each time.

We normally use a public final field for the top and bottom annotations in a type system. So, please just make the field public and final.